python/gtsam/tests/test_Cal3Unified.py

```diff
@@ -23,6 +23,89 @@ def test_Cal3Unified(self):
         self.assertEqual(K.fx(), 1.)
         self.assertEqual(K.fx(), 1.)
 
+    def test_distortion(self):
+        "Stereographic fisheye model of focal length f, defined as r/f = 2*tan(theta/2)"
+        fx, fy, s, u0, v0 = 2, 2, 0, 0, 0
+        k1, k2, p1, p2 = 0, 0, 0, 0
+        xi = 1
+        stereographic = gtsam.Cal3Unified(fx, fy, s, u0, v0, k1, k2, p1, p2, xi)
+        x, y, z = 1, 0, 1
+        u, v = stereographic.uncalibrate([x, y])
+        r = np.linalg.norm([x, y, z])
+        # Note: 2*tan(atan2(x, z)/2) = 2/(1+sqrt(x^2+z^2))
+        self.assertAlmostEqual(2*np.tan(np.arctan2(x, z)/2), 2/(1+r))
+        self.assertAlmostEqual(u, 2/(1+r))
+        x2, y2 = stereographic.calibrate([u, v])
+        self.assertAlmostEqual(x2, x)
+
+    def test_pinhole(self):
+        "Spatial stereographic camera projection"
+        x, y, z = 1.0, 0.0, 1.0
+        r = np.linalg.norm([x, y, z])
+        u, v = 2/(1+r), 0.0
+        objPoint = np.array([x, y, z])
+        imgPoint = np.array([u, v])
+        fx, fy, s, u0, v0 = 2, 2, 0, 0, 0
+        k1, k2, p1, p2 = 0, 0, 0, 0
+        xi = 1
+        stereographic = gtsam.Cal3Unified(fx, fy, s, u0, v0, k1, k2, p1, p2, xi)
+        pose = gtsam.Pose3()
+        camera = gtsam.PinholeCameraCal3Unified(pose, stereographic)
+        pt1 = camera.Project(objPoint)
+        self.gtsamAssertEquals(pt1, np.array([x/z, y/z]))
+        pt2 = camera.project(objPoint)
+        self.gtsamAssertEquals(pt2, np.array([u, v]))
+        obj1 = camera.backproject([u, v], z)
+        self.gtsamAssertEquals(obj1, np.array([x, y, z]))
+        r1 = camera.range(np.array([x, y, z]))
+        self.assertEqual(r1, r)
+
+    def test_generic_factor(self):
+        "Evaluate residual using pose and point as state variables"
+        fx, fy, s, u0, v0 = 2, 2, 0, 0, 0
+        k1, k2, p1, p2 = 0, 0, 0, 0
```

Comment:
Put all these set up values in a `setUp` function for the class and just access the variables directly?